plume-file-core/src/main/java/com/coreoz/plume/file/services/file/FileServiceDisk.java

```diff
@@ -23,42 +11,65 @@
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.io.File;
+import java.io.IOException;
+import java.util.Optional;
 
-// TODO should be unit tested
+// TODO file creation should be unit tested
+@Singleton
 public class FileServiceDisk implements FileService {
     private static final Logger logger = LoggerFactory.getLogger(FileServiceDisk.class);
-    protected String path;
-    private String baseUrl;
+
+    private final String path;
+    private final String baseUrl;
+    private final String fileWsPath;
     private final FileDaoDiskQuerydsl fileDao;
     private final FileTypesProvider fileTypesProvider;
     private final ChecksumService checksumService;
 
     @Inject
-    public FileServiceDisk(FileDaoDiskQuerydsl fileDao,
-                           FileTypesProvider fileTypesProvider,
-                           FileConfigurationService configurationService,
-                           ChecksumService checksumService) {
-        path = configurationService.mediaLocalPath();
-        baseUrl = configurationService.apiBasePath() + "/";
+    public FileServiceDisk(
+        FileDaoDiskQuerydsl fileDao,
+        FileTypesProvider fileTypesProvider,
+        FileConfigurationService configurationService,
+        ChecksumService checksumService
+    ) {
+        this.path = configurationService.mediaLocalPath();
+        this.baseUrl = configurationService.apiBasePath();
+        this.fileWsPath = configurationService.fileWsPath();
         this.fileDao = fileDao;
         this.fileTypesProvider = fileTypesProvider;
         this.checksumService = checksumService;
     }
 
     @Override
-    public FileUploaded upload(FileType fileType, byte[] fileData, @Nullable String filename) {
-        String fileName = FileNameUtils.sanitize(Strings.nullToEmpty(filename));
-
-        String relativePath = String.valueOf(IdGenerator.generate()) + extractFileExtension(fileName);
+    public FileUploaded upload(FileType fileType, byte[] fileData, String filename) {
+        if (Strings.isNullOrEmpty(filename)) {
+            throw new RuntimeException("File name must be declared to be saved on disk");
+        }
+        String fileName = FileNameUtils.sanitize(filename);
 
-        FileEntityQuerydsl file = fileDao.upload(fileType.name(), fileName, relativePath);
+        FileEntryDisk file = this.fileDao.upload(fileType.name(), fileName, fileName);
 
-        createFile(path, fileData, relativePath);
+        this.createFile(
+            this.path,
+            fileData,
+            file.getPath()
+        );
 
-        return FileUploaded.of(file.getId(), url(file.getId(), fileName));
+        return FileUploaded.of(
+            file.getId(),
+            file.getUid(),
+            url(file.getUid(), fileName)
+        );
     }
 
-    private void createFile(String path, byte[] fileData, @Nullable String fileName) {
+    private void createFile(String path, byte[] fileData, String fileName) {
```

Comment:
La création du fichier devrait être dans un sous dossier par type de fichier

Reply:
Cf `FileType` pour le type de fichier

Reply:
et si on enregistre en db, le champ spécifiant le dossier dans FileType serait à null ?
Ou bien le chemin serrait base-path/{FileType.name()}/ ?

Reply:
Le commentaire est que pour l'enregistrement sur le disque.
Pour la partie DB rien ne change.

On aurait bien les images dans `base-path/{FileType.name()}/`

Reply:
yes noté